.changeset/giant-bees-speak.md

```diff
@@ -0,0 +1,9 @@
+---
+'@backstage/plugin-techdocs-backend': minor
+---
+
+**BREAKING**: The `cache` option is now required by `createRouter`.
+
+Added catalog-based authorization to TechDocs backend. When permissions are enabled for Backstage (via the `permission.enabled` config) the current user must have read access to the doc's corresponding catalog entity. The backend will return a 404 if the current user doesn't have access.
+
+Note: Custom TechDocs setups that expose docs without a corresponding catalog entity will not work with permissions enabled, but a solution for this will be provided in the future.
```

Comment:
Maybe a more generic phrasing of this; docs without linked catalog entities may be one use-case, but there are likely others.  Perhaps:

> Note: If you publish your TechDocs documentation under custom paths that do not conform to the default `:namespace/:kind/:name` pattern, TechDocs will not work with permissions enabled....

Reply:
Good point! I also added a note about joining the discord to help us better understand these custom use cases. I saw some other changesets link to discord so I figured that was okay 😄